pkg/work/spoke/conditions/reader.go

```diff
@@ -0,0 +1,215 @@
+		message, err = s.getConditionMessageByRule(obj, rule, status == metav1.ConditionTrue)
+		if err != nil && message == "" {
+			message = err.Error()
+		}
+	}
+
+	return metav1.Condition{
+		Type:    rule.Condition,
+		Status:  status,
+		Message: message,
+		Reason:  reason,
+	}, err
+}
+
+func (s *ConditionReader) evaluateCelExpressions(obj *unstructured.Unstructured, expressions []string) (metav1.ConditionStatus, string, error) {
+	env, err := s.celEnv(cel.Variable("object", cel.DynType))
+	if err != nil {
+		return metav1.ConditionUnknown, workapiv1.ConditionRuleInternalError, err
+	}
+
+	for _, expression := range expressions {
+		ast, iss := env.Compile(expression)
+		err = iss.Err()
+		if err != nil {
+			return metav1.ConditionFalse, workapiv1.ConditionRuleExpressionError, err
+		}
+
+		prg, err := env.Program(ast)
+		if err != nil {
+			return metav1.ConditionUnknown, workapiv1.ConditionRuleInternalError, err
+		}
+
+		out, _, err := prg.Eval(map[string]any{
+			"object": obj.Object,
+		})
+		if err != nil {
+			return metav1.ConditionFalse, workapiv1.ConditionRuleExpressionError, err
+		}
+
+		switch result := out.Value().(type) {
+		case bool:
+			// If result is false return early
+			// Otherwise continue checking expressions
+			if !result {
+				return metav1.ConditionFalse, workapiv1.ConditionRuleEvaluated, nil
+			}
+		default:
+			err = fmt.Errorf("expected bool result, got %v", reflect.TypeOf(result))
+			return metav1.ConditionFalse, workapiv1.ConditionRuleExpressionError, err
+		}
+	}
+
+	return metav1.ConditionTrue, workapiv1.ConditionRuleEvaluated, nil
+}
+
+func (s *ConditionReader) getConditionMessageByRule(obj *unstructured.Unstructured, rule workapiv1.ConditionRule, result bool) (string, error) {
+	if rule.MessageExpression != "" {
+		env, err := s.celEnv(cel.Variable("object", cel.DynType), cel.Variable("result", cel.BoolType))
+		if err != nil {
+			return "", err
+		}
+
+		ast, iss := env.Compile(rule.MessageExpression)
+		err = iss.Err()
+		if err != nil {
+			// Trim CEL code snippets out of message
+			message := strings.Split(iss.String(), "\n | ")[0]
+			return message, err
+		}
+
+		prg, err := env.Program(ast)
+		if err != nil {
+			return "", err
+		}
+
+		out, _, err := prg.Eval(map[string]any{
+			"object": obj.Object,
+			"result": result,
+		})
+		if err != nil {
+			return "", err
+		}
+
+		switch message := out.Value().(type) {
+		case string:
+			if message != "" {
+				return message, nil
+			}
+		default:
+			return "", fmt.Errorf("expected message expression to have a string result, got %v", reflect.TypeOf(message))
+		}
+	}
+
+	if rule.Message != "" {
+		return rule.Message, nil
+	}
+	if result {
+		return "Manifest is " + rule.Condition, nil
+	}
+	return "Manifest is not " + rule.Condition, nil
+}
+
+func (s *ConditionReader) celEnv(opts ...cel.EnvOption) (*cel.Env, error) {
+	opts = slices.Concat(
+		opts,
+		common.BaseEnvOpts,
+		[]cel.EnvOption{
+			cel.Function("hasConditions", cel.Overload(
```

Comment:
Should this be put into sdk-go library? Some description of the usage and an example may help reader better unstand how this function works.

Reply:
Do you mean the hasConditions function? Yes I'd be happy to add it to sdk-go.  I found checking for the existence of status conditions to be rather verbose in raw CEL, so it could certainly be useful elsewhere.